compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1478,7 +1478,9 @@ impl<'tcx> Place<'tcx> {
     /// It's guaranteed to be in the first place
     pub fn has_deref(&self) -> bool {
         // To make sure this is not accidently used in wrong mir phase
-        debug_assert!(!self.projection[1..].contains(&PlaceElem::Deref));
+        debug_assert!(
+            self.projection.is_empty() || !self.projection[1..].contains(&PlaceElem::Deref)
+        );
```

Comment:
Oh, so *that* was the actual cause of the strange ICE...
Cc @ouz-a

Reply:
I wonder how this slipped without causing any issues until now.

Reply:
Looks like the ui test suite never hits the code path where this is empty.